src/mscorlib/src/System/Buffer.cs

```diff
@@ -545,47 +538,62 @@ internal unsafe static void Memmove(byte* dest, byte* src, nuint len)
 #if BIT64
                 *(long*)(dest + i) = *(long*)(src + i);
                 *(long*)(dest + i + 8) = *(long*)(src + i + 8);
-#else
+#else // BIT64
                 *(int*)(dest + i) = *(int*)(src + i);
                 *(int*)(dest + i + 4) = *(int*)(src + i + 4);
                 *(int*)(dest + i + 8) = *(int*)(src + i + 8);
                 *(int*)(dest + i + 12) = *(int*)(src + i + 12);
-#endif
+#endif // BIT64
 
                 i = counter;
-                
+
                 // See notes above for why this wasn't used instead
                 // i += 16;
             }
             while (counter <= end);
 
-            if ((len & 8) != 0)
+            // We have <= 15 bytes to copy at this point.
+
+            if ((mask & 8) != 0)
```

Comment:
Excellent idea, we can use a switch statement here to avoid the branches/unaligned write (and since code size doesn't matter, as all this is crossgened and there is no runtime overhead for larger methods). I'm going to experiment with this and see if I can tease out some even better perf.